app/assets/javascripts/discourse/app/lib/plugin-api.gjs

```diff
@@ -1803,49 +1804,95 @@ class PluginApi {
   /**
    * Allows adding icons to the category-link html
    *
-   * ```
+   * ```javascript
    * api.addCategoryLinkIcon((category) => {
-   *  if (category.someProperty) {
-        return "eye"
-      }
+   *   if (category.someProperty) {
+   *     return "eye"
+   *   }
    * });
    * ```
-   *
    **/
   addCategoryLinkIcon(renderer) {
     addExtraIconRenderer(renderer);
   }
+
   /**
-   * Adds a widget or a component to the header-icon ul.
+   * Allows for manipulation of the header icons. This includes, adding, removing, or modifying the order of icons.
    *
-   * If adding a widget it must already be created. You can create new widgets
-   * in a theme or plugin via an initializer prior to calling this function.
+   * Only the passing of components is supported, and by default the icons are added to the right of exisiting icons.
    *
+   * Example: Add the chat icon to the header icons before the search icon
    * ```
-   * api.addToHeaderIcons(
-   *  createWidget("some-widget")
+   * api.headerIcons.add(
+   *  "chat",
+   *  ChatIconComponent,
+   *  { before: "search" }
+   * )
+   * ```
+   *
+   * Example: Remove the chat icon from the header icons
+   * ```
+   * api.headerIcons.delete("chat")
+   * ```
+   *
+   * Example: Reposition the chat icon to be before the user-menu icon and after the hamburger icon
+   * ```
+   * api.headerIcons.reposition("chat", { before: "user-menu", after: "hamburger" })
+   * ```
+   *
+   * Example: Check if the chat icon is present in the header icons (returns true of false)
+   * ```
+   * api.headerIcons.has("chat")
    * ```
    *
-   * If adding a component you can pass the component directly. Additionally, you can
-   * utilize the `@panelPortal` argument to create a dropdown panel. This can be useful when
+   * Additionally, you can utilize the `@panelPortal` argument to create a dropdown panel. This can be useful when
    * you want create a button in the header that opens a dropdown panel with additional content.
    *
    * ```
-   * api.addToHeaderIcons(
-      <template>
-        <span>Icon</span>
+   * const IconWithDropdown = <template>
+   *   <DButton @icon="icon" @onClick={{this.toggleVisible}} />
+   *   {{#if this.visible}}
+   *     <@panelPortal>
+   *       <div>Panel</div>
+   *     </@panelPortal>
+   *   {{/if}}
+   * </template>;
+   *
+   * api.headerIcons.add("icon-name", IconWithDropdown, { before: "search" })
+   * ```
+   *
+   **/
+  get headerIcons() {
+    const currentUser = this._lookupContainer("service:current-user");
+    return currentUser?.glimmer_header_enabled
+      ? headerIconsDAG()
+      : widgetHeaderIconsDAG();
```

Comment:
Hmm this is not gonna work if we go down the 'automatic feature flag' route. (because we'll want to defer the decision about the flag until after the app has initialized)

Could we make the widget implementation import the DAG from the glimmer version? And then ignore the ones named 'hamburger', 'user' and 'search' ?